extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php

```diff
@@ -0,0 +1,328 @@
+ *
+ * @author Sean Colombo
+ */
+class AppPromoLandingController extends WikiaController {
+	
+	private static $extraBodyClasses = []; // TODO: REMOVE - ONLY DURING THE TEMPORARY COPYING OF OASISCONTROLLER
+
+	const RESPONSE_OK = 200;
+	const APP_CONFIG_SERVICE_URL = "http://prod.deploypanel.service.sjc.consul/api/app-configuration/";
+	const BRANCH_API_URL = "https://api.branch.io/v1/app/";
+
+	// Settings for the background image-grid.
+	const MAX_TRENDING_ARTICLES = 40; // we need about 33 images, and not all articles have images.
+	const THUMBS_PER_ROW = 11;
+	const THUMBS_NUM_ROWS = 3;
+	const IMG_HEIGHT = 184; // sizes directly from Zeplin.io mockup.
+	const IMG_WIDTH = 184;
+
+	protected static $CACHE_KEY = "mobileAppConfigs";
+	protected static $CACHE_KEY_BRANCH = "branchioBranchKey";
+	protected static $CACHE_KEY_VERSION = 0.1;
+	protected static $CACHE_KEY_VERSION_BRANCH = 0.1;
+	protected static $CACHE_EXPIRY = 86400;
+
+	/**
+	 * Render HTML for whole App Promo Landing page.
+	 */
+	public function executeIndex() {
+		wfProfileIn(__METHOD__);
+
+		// Since this "Community_App" article won't be found, we need to manually say it's okay so that it's not a 404.
+		$this->response->setCode(self::RESPONSE_OK);
+		
+		// Pull in the app-configuration (has data for all apps)
+		$appConfig = [];
+		$memcKey = wfMemcKey( static::$CACHE_KEY, static::$CACHE_KEY_VERSION );
+		$response = F::app()->wg->memc->get( $memcKey );
+		if ( empty( $response ) ){
+			$req = MWHttpRequest::factory( self::APP_CONFIG_SERVICE_URL, array( 'noProxy' => true ) );
+			$status = $req->execute();
+			if( $status->isOK() ) {
+				$response = $req->getContent();
+				if ( empty( $response ) ) {
+					wfProfileOut( __METHOD__ );
+					throw new EmptyResponseException( self::APP_CONFIG_SERVICE_URL );
+				} else {
+					// Request was successful. Cache it in memcached (faster than going over network-card even on our internal network).
+					F::app()->wg->memc->set( $memcKey, $response, static::$CACHE_EXPIRY );
+				}
+			}
+		}
+
+		$appConfig = json_decode( $response );
+		if(empty($appConfig)){
+
+			// TODO: How should we handle the error of not having an appConfig? We won't be able to link the user to the apps.
+
+		}
+
+		$config = $this->getConfigForWiki($appConfig, $this->wg->CityId);
+
+		// Create the direct-link URLs for the apps on each store.
+		$this->androidUrl = $this->getAndroidUrl($config);
+		$this->iosUrl = $this->getIosUrl($config);
+
+		// Inject the JS
+		$srcs = AssetsManager::getInstance()->getGroupCommonURL( 'app_promo_landing_js' );
+		foreach( $srcs as $src ) {
+			$this->wg->Out->addScript( "<script type=\"{$wgJsMimeType}\" src=\"{$src}\"></script>" );
+		}
+
+		// Inject the JS
+		$srcs = AssetsManager::getInstance()->getGroupCommonURL( 'app_promo_landing_js' );
+		foreach( $srcs as $src ) {
+			$this->wg->Out->addScript( "<script type=\"{$wgJsMimeType}\" src=\"{$src}\"></script>" );
+		}
```

Comment:
I removed one in a commit that is now in the PR, but it detected this one as still being here, so this comment is still visible, but the problem is resolved :+1: